contrib/drivers/volume/huawei/oceanstor/oceanstor.go

```diff
@@ -21,27 +21,33 @@ import (
 	"strings"
 
 	log "github.com/golang/glog"
+	"github.com/opensds/opensds/contrib/drivers/factory"
 	. "github.com/opensds/opensds/contrib/drivers/utils/config"
+	. "github.com/opensds/opensds/contrib/drivers/utils/constants"
 	"github.com/opensds/opensds/pkg/model"
 	pb "github.com/opensds/opensds/pkg/model/proto"
 	"github.com/opensds/opensds/pkg/utils"
 	"github.com/opensds/opensds/pkg/utils/config"
 	uuid "github.com/satori/go.uuid"
 )
 
+func NewDriver(properties config.BackendProperties) factory.Driver {
+	return &Driver{properties: properties}
+}
+
 type Driver struct {
-	conf   *OceanStorConfig
-	client *OceanStorClient
+	properties config.BackendProperties
+	conf       *OceanStorConfig
+	client     *OceanStorClient
 }
 
 func (d *Driver) Setup() (err error) {
 	// Read huawei oceanstor config file
 	conf := &OceanStorConfig{}
 	d.conf = conf
-	path := config.CONF.OsdsDock.Backends.HuaweiOceanStorBlock.ConfigPath
-
-	if "" == path {
-		path = defaultConfPath
+	path := defaultConfPath
```

Comment:
this piece of code is repeated in all drivers...please add to base class/struct (DriverBase) or similar